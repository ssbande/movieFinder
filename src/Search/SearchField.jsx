import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { Input } from 'semantic-ui-react';

const SearchField = ({ query, handleChange }) => {
  const inputRef = useRef(null);

  useEffect(() => inputRef.current.focus(), []);

  return (
    <Input className='searchBox' onChange={handleChange} ref={inputRef} fluid
      placeholder='Search a Movie, TV Show or Artist ...'
      action={{
        disabled: !query.length,
        color: query.length ? 'teal' : 'grey',
        icon: 'search',
        content: 'Search',
      }}
    />
  )
}

SearchField.propTypes = {
  query: PropTypes.string,
  handleChange: PropTypes.func.isRequired
}

export default SearchField;