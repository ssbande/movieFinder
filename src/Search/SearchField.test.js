import React from 'react';
import { shallow } from 'enzyme';

import SearchField from './SearchField';

const setup = () => ({
  props: {
    query: 'searchString',
    handleChange: jest.fn(),
  }
})

describe('Input field for search bar', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<SearchField {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})