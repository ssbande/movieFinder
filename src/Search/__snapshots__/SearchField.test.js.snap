// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field for search bar should render correctly 1`] = `ShallowWrapper {}`;
